From 04e6139d52aeef4c68f79d33662d2f3832fd993c Mon Sep 17 00:00:00 2001
From: Ondrej Kubik <ondrej.kubik@canonical.com>
Date: Fri, 29 Apr 2022 20:13:32 +0100
Subject: [PATCH 3/5] compat.c: configure correct available shells

Signed-off-by: Ondrej Kubik <ondrej.kubik@canonical.com>
---
 compat.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/compat.c b/compat.c
index 7a0e78a..9cdcdb7 100644
--- a/compat.c
+++ b/compat.c
@@ -82,6 +82,7 @@
 
 #include "includes.h"
 
+#undef HAVE_GETUSERSHELL
 #ifndef HAVE_GETUSERSHELL
 static char **curshell, **shells, *strings;
 static char **initshells();
@@ -232,7 +233,7 @@ void setusershell() {
 
 static char **initshells() {
 	/* don't touch this list. */
-	static const char *okshells[] = { "/bin/sh", "/bin/csh", NULL };
+	static const char *okshells[] = { "/bin/sh", "/bin/bash", "/bin/rbash", "/bin/dash", NULL };
 	register char **sp, *cp;
 	register FILE *fp;
 	struct stat statb;
-- 
2.25.1

