diff --git a/common-session.c b/common-session.c
index aa31e49..adcdee7 100644
--- a/common-session.c
+++ b/common-session.c
@@ -623,7 +623,7 @@ void fill_passwd(const char* username) {
 	ses.authstate.pw_uid = pw->pw_uid;
 	ses.authstate.pw_gid = pw->pw_gid;
 	ses.authstate.pw_name = m_strdup(pw->pw_name);
-	ses.authstate.pw_dir = m_strdup(pw->pw_dir);
+	ses.authstate.pw_dir = getenv("SNAP_COMMON");// m_strdup(pw->pw_dir);
 	ses.authstate.pw_shell = m_strdup(pw->pw_shell);
 	{
 		char *passwd_crypt = pw->pw_passwd;
diff --git a/default_options.h b/default_options.h
index 9000fcc..564ea65 100644
--- a/default_options.h
+++ b/default_options.h
@@ -38,7 +38,7 @@ IMPORTANT: Some options will require "make clean" after changes */
 
 /* Include verbose debug output, enabled with -v at runtime. 
  * This will add a reasonable amount to your executable size. */
-#define DEBUG_TRACE 0
+#define DEBUG_TRACE 1
 
 /* Set this if you want to use the DROPBEAR_SMALL_CODE option. This can save
  * several kB in binary size however will make the symmetrical ciphers and hashes
diff --git a/svr-auth.c b/svr-auth.c
index 7575f90..c54e3ba 100644
--- a/svr-auth.c
+++ b/svr-auth.c
@@ -321,13 +321,13 @@ static int checkusername(const char *username, unsigned int userlen) {
 	 * should return some standard shells like "/bin/sh" and "/bin/csh" (this
 	 * is platform-specific) */
 	setusershell();
-	while ((listshell = getusershell()) != NULL) {
-		TRACE(("test shell is '%s'", listshell))
-		if (strcmp(listshell, usershell) == 0) {
+//	while ((listshell = getusershell()) != NULL) {
+//		TRACE(("test shell is '%s'", listshell))
+//		if (strcmp(listshell, usershell) == 0) {
 			/* have a match */
 			goto goodshell;
-		}
-	}
+//		}
+//	}
 	/* no matching shell */
 	endusershell();
 	TRACE(("no matching shell"))
